lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -335,3 +376,26 @@ function dosomething_rogue_check_rbid_and_store_ref($rogue_reportback) {
 
    return $data;
  }
+
+/**
+ * Query to check if a reportback item exists in the dosomething_rogue_failed_task_log table.
+ *
+ * @param string $northstar_id
+ * @param int $campaign_run_id
+ * @param string $caption
+ * @param string $file
+ *
+ */
+function dosomething_rogue_rb_previously_failed($northstar_id, $campaign_run_id, $caption, $file) {
+  return db_query("SELECT * FROM {dosomething_rogue_failed_task_log} failed_rbs WHERE northstar_id = :northstar_id AND campaign_run_id = :campaign_run_id AND caption = :caption AND file = :file", array(':northstar_id' => $northstar_id, ':campaign_run_id' => $campaign_run_id, ':caption' => $caption, ':file' => $file))->fetchAll();
+}
+
+/**
+ * Query to check if a reportback item failed updated status exists in the dosomething_rogue_failed_task_log table.
+ *
+ * @param int $rogue_item_id
+ *
+ */
+function dosomething_rogue_rb_item_previously_failed($rogue_item_id) {
```

Comment:
Is there a way you can combine `dosomething_rogue_rb_previously_failed()` and `dosomething_rogue_rb_item_previously_failed($rogue_item_id)` If you generalize the params you pass in and also pass in like the `type` maybe you could make a more general query that just checks if something is already in the table. ?